### PR TITLE
add ability to skip files

### PR DESCRIPTION
**Pull request by kentcdodds**
Lots of this was borrowed from https://github.com/adana-coverage/babel-plugin-transform-adana :-) OSS

friends-thumbs-up-matthew-perry-JPsFUPp3vLS5q

### CODE REVIEW

**Comment by kentcdodds** [comment]
Unfortunately, I'm not sure how to test this functionality :-/

**Comment by kentcdodds** [comment]
But, I should say that I tried it out locally in my project and it worked great!

**Comment by kentcdodds** [comment]
Just realized there's a section in the README that would need to be removed when this is merged. I'm going to sleep now, but if you want to wait to merge this until I've done that, I'll give it a look tomorrow (or you can just do it yourself after merging.... It's your project :smile:)

**Comment by kentcdodds** [comment]
I updated the PR to remove the section in the README about ignoring files.

**Comment by dtinth** [comment]
Hello. Thanks for your patch! :+1: 

- Could you please run the code through `./node_modules/.bin/standard`? Sorry I haven’t put it in Travis. :eyeglasses: 

- Maybe it would be nice if you could link to the original source code in the comment to give credit where it is due. It seems like the the code is copied verbatim from there. :smile: 

- To test whether a file is covered or not, you can create a [fixture](https://github.com/dtinth/babel-plugin-__coverage__/tree/5b4e9cbe506e3f15ae106b1aa788daa9adfff898/fixtures) and ignore it by specifying [plugin options in the test](https://github.com/dtinth/babel-plugin-__coverage__/blob/5b4e9cbe506e3f15ae106b1aa788daa9adfff898/test.js#L11). Then you can validate that `global.__coverage__[require.resolve('../fixtures/skip')] === undefined` after requiring. :sparkles:

- Don’t worry about the coverage in lines 41 and 50. I think that’s just-in-case for super weird case (where it is better to just cover the file rather than bailing out). But if you really want _moar coverage!!_ then you can extract `skip` into its own file and try calling `skip()` from the test. That should cover the super weird case. :100:

**Comment by kentcdodds** [comment]
Updated with all of your feedback. I agree that it doesn't make sense to cover lines 41 and 50 :) But it's also nice to have it there so we don't get a bug report months down the road when something weird happens :-)

File: test.js [hunk 1]
```diff
@@ -100,6 +100,28 @@ describe('test', function () {
     it('works with label statement', function () {
       assert.equal(require('./fixtures/label'), 1)
     })
+
+    it('ignores files excluded by `ignore` option', function () {
+      assert.ok(global.__coverage__[require.resolve('./fixtures/ignored')] === undefined)
```
Comment (kentcdodds) [inline test.js]: Wondering if it'd be better to test this one the same way the other two tests below are being tested and to extract this logic into a function to be shared. Thoughts?

**Comment by kentcdodds** [comment]
Updated again. I think this is a bit cleaner :+1:

**Comment by kentcdodds** [comment]
Added a few more tests just to cover bases and be more descriptive of how things work. Let me know if anything's missing :+1: I'm done now, I promise.

File: test.js [hunk 2]
```diff
@@ -132,4 +157,16 @@ describe('test', function () {
...
+  function transpileFile (filename, pluginOptions) {
+    return Babel.transformFileSync(require.resolve(filename), {
+      presets: [ 'es2015' ],
+      plugins: [ [ __coverage__Plugin, pluginOptions ] ],
+      babelrc: false
+    }).code
+  }
+
+  function codeIsCovered (code) {
+    return code.indexOf('_cover__') !== -1
```
Comment (dtinth) [inline test.js]: Nice trick! :100:

**Comment by dtinth** [comment]
I just reviewed it, that’s great! Thanks for contributing; it’s very much appreciated. :tada:

**Comment by dtinth** [comment]
Published as [v0.11111.0](https://github.com/dtinth/babel-plugin-__coverage__/tree/v0.11111.0). :ship:

**Pull request merged**